pkg/systempreferences/dock.go

```diff
@@ -33,11 +35,39 @@ func setDockApps(apps []string) {
 	defaultsDelete("com.apple.dock", "persistent-apps")
 
 	for _, app := range apps {
-		xmlApp := fmt.Sprintf("<dict><key>tile-data</key><dict><key>file-data</key><dict><key>_CFURLString</key><string>%s</string><key>_CFURLStringType</key><integer>0</integer></dict></dict></dict>", app)
+		appPath, err := getAppPath(app)
+
+		if err != nil {
+			log.Fatalf("getAppPath failed for %s\n", err)
+		}
+
+		xmlApp := fmt.Sprintf("<dict><key>tile-data</key><dict><key>file-data</key><dict><key>_CFURLString</key><string>%s</string><key>_CFURLStringType</key><integer>0</integer></dict></dict></dict>", appPath)
 		defaultsArrayAdd("com.apple.dock", "persistent-apps", xmlApp)
 	}
 }
 
+func getAppPath(appName string) (string, error) {
+	cmd := exec.Command("mdfind", "-name", appName)
+
+	out, err := cmd.Output()
+
+	if err != nil {
+		log.Fatalf("cmd.Run() failed with %s\n", err)
```

Comment:
you could return here instead:
```suggestion
		return "", err
```